Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs

```diff
@@ -434,6 +434,27 @@ eval (IsMockLocation isMock) state = do
       _ = unsafePerformEffect $ if val then  logEvent (state.data.logField) "ny_fakeGPS_enabled" else pure unit -- we are using unsafePerformEffect becasue without it we are not getting logs in firebase, since we are passing a parameter from state i.e. logField then the output will be inline and it will not be able to precompute so it's safe to use it here.
   continue state{props{isMockLocation = false}}
 
+eval (UpdateCurrentStageStatus stage (RideBookingStatusRes resp)) newState = do
```

Comment:
This function looks little messy, is it properly tested?

Reply:
this function is same as the earlier one...only the diff is of the response Type...rest if else all conditions are same as earlier one...that can be replaced with better case statement and guard operator..but it was there in earlier code and was working fine so decided not to change anything.
for testing: 
- i did one rideFlow
- killed the app and restarted as expected it called the rideList api and then kept poling the new status api.
- changes screens and came back at home to check the status
- completed one rideFlow.